src/graph/GoExecutor.cpp

```diff
@@ -75,6 +79,13 @@ void GoExecutor::execute() {
         onEmptyInputs();
         return;
     }
+    if (distinct_) {
+        std::unordered_set<VertexID> uniqID;
+        for (auto id : starts_) {
+            uniqID.emplace(id);
+        }
+        starts_ = std::vector<VertexID>(uniqID.begin(), uniqID.end());
```

Comment:
what if distinct & order by?

Reply:
Order by is standalone statement, they will not affect each other.